docs/DeveloperGuide.md

```diff
@@ -283,31 +283,297 @@ Priorities: High (must have) - `* * *`, Medium (nice to have) - `* *`, Low (unli
+(For all use cases below, the **System** is the `Address Book` and the **Actor** is the `user`, unless specified otherwise)
 
-**Use case: Delete a person**
+**Use case: UC01 - Add a new contact**
 
 **MSS**
 
-1.  User requests to list persons
-2.  AddressBook shows a list of persons
-3.  User requests to delete a specific person in the list
-4.  AddressBook deletes the person
+1. User requests to add a new contact to the Address Book.
+2. Address Book adds contact to the contact list.
 
-    Use case ends.
+   Use case ends.
+
+**Extensions**
+
+* 1a. The given input is invalid.
+
+    * 1a1. Address Book shows an error message.
+
+      Use case ends.
+
+**Use case: UC02 - Update a contact**
+
+**MSS**
+
+1. User requests to update details for a person in the Address Book.
+2. Address Book updates the details of the contact.
+
+   Use case ends.
+
+**Extensions**
+
+* 1a. The given input is invalid.
+
+    * 1a1. Address Book shows an error message.
+
+      Use case ends.
+
+* 1b. The given index is invalid.
+
+    * 1b1. Address Book shows an error message.
+
+      Use case ends.
+
+**Use case: UC03 - List all contacts in Address Book**
+
+**MSS**
+
+1. User requests to list all contacts.
+2. Address Book shows a list of contacts if any.
+
+   Use case ends.
+
+**Use case: UC04 - Delete a contact**
+
+**MSS**
+
+1. User requests to list all contacts (UC03).
+2. Address Book shows a list of contacts (UC03).
+3. User requests to delete a specific contact in the list.
+4. Address Book deletes the contact.
+
+   Use case ends.
 
 **Extensions**
 
 * 2a. The list is empty.
 
   Use case ends.
 
-* 3a. The given index is invalid.
+* 3a. The given contact is invalid.
 
     * 3a1. AddressBook shows an error message.
 
       Use case resumes at step 2.
 
+**Use case: UC05 - Find contact by keyword**
+
+**MSS**
+
+1. User requests to find a person.
+2. Address Book shows a list of persons with names containing the keyword, if any.
+
+   Use case ends.
+
+**Extensions**
+
+* 1a. No keyword is given.
+
+    * 1a1. Address Book shows an error message.
+
+      Use case ends.
+
+**Use case: UC06 - Set previously contacted date**
+
+**MSS**
+
+1. User request to set previously contacted date for a specified contact.
+2. Address Book sets the previously contacted date for the specified contact.
+
+   Use case ends.
+
+**Extensions**
+
+* 1a. No date is given.
+
+    * 1a1. Address Book sets the previously contacted date as today for the specified contact.
+
+      Use case ends.
+
+* 1b. The given date is invalid.
+
+    * 1b1. Address Book shows an error message.
+
+      Use case ends.
+
+* 1c. The given index is invalid.
+
+    * 1c1. Address Book shows an error message.
+
+      Use case ends.
+
+**Use case: UC07 - List all contacts in Address Book within a designated time frame**
+
+**MSS**
+
+1. User requests to see all contacts last contacted within a designated number of days.
```

Comment:
Add an extension for a case whereby there are no contacted clients within the designated number of days